### ISSUE TITLE
using stack to build a github-release binary fails with 404

### ISSUE THREAD

**Issue by dankamongmen** [tfausak/github-release#9]
Hey there. I'm interested in using github-release, but attempting to use `stack` as outlined in the README fails:

```
[schwarzgerat](0) $ stack --resolver nightly install github-release
Selected resolver: nightly-2020-07-24
Downloading nightly-2020-07-24 build plan ...
RedownloadInvalidResponse Request {
  host                 = "raw.githubusercontent.com"
  port                 = 443
  secure               = True
  requestHeaders       = []
  path                 = "/fpco/stackage-nightly/master//nightly-2020-07-24.yaml"
  queryString          = ""
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 "/home/dank/.stack/build-plan/nightly-2020-07-24.yaml" (Response {responseStatus = Status {statusCode = 404, statusMessage = "Not Found"}, responseVersion = HTTP/1.1, responseHeaders = [("Connection","keep-alive"),("Content-Length","14"),("Content-Security-Policy","default-src 'none'; style-src 'unsafe-inline'; sandbox"),("Strict-Transport-Security","max-age=31536000"),("X-Content-Type-Options","nosniff"),("X-Frame-Options","deny"),("X-XSS-Protection","1; mode=block"),("Content-Type","text/plain; charset=utf-8"),("Via","1.1 varnish (Varnish/6.0)"),("X-GitHub-Request-Id","E728:34C4:1B5A5E:213919:5F1D0505"),("Accept-Ranges","bytes"),("Date","Sun, 26 Jul 2020 04:22:30 GMT"),("Via","1.1 varnish"),("X-Served-By","cache-pdk17883-PDK"),("X-Cache","MISS, MISS"),("X-Cache-Hits","0, 0"),("X-Timer","S1595737351.558432,VS0,VE73"),("Vary","Authorization,Accept-Encoding"),("Access-Control-Allow-Origin","*"),("X-Fastly-Request-ID","e5ba0d3ab55daf4a9983620bcf3ecf7a4cc6ca06"),("Expires","Sun, 26 Jul 2020 04:27:30 GMT"),("Source-Age","0")], responseBody = (), responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose})
[schwarzgerat](1) $ 
```

I've never used `stack`, and am not particularly skilled with Haskell, so I'm at something of a loss on how to proceed.

this is `haskell-stack` as packaged in Debian Unstable, 1.9.3.1-1. tip of trunk for github-release.

**Comment by tfausak** [comment]
Huh, that's weird. `github-release` is definitely in the `nightly-2020-07-24` snapshot: https://www.stackage.org/nightly-2020-07-24/package/github-release-1.3.4

It looks like Stack is failing to download the snapshot. Can you try upgrading Stack with `stack upgrade`? Version 1.9.3 is a little old. The latest is 2.3.1. Maybe they changed something with how snapshots work behind the scenes.

**Comment by dankamongmen** [comment]
Sure. I ended up using [githubrelease](https://pypi.org/project/githubrelease/) from pypi, but I'm happy to assist any debugging.

(with that said, I wouldn't typically want to go outside my package manager unless there was particularly good reason to)

```
[schwarzgerat](130) $ sudo stack upgrade
Current Stack version: 1.9.3.1, available download version: 2.3.1
Newer version detected, downloading
Querying for archive location for platform: linux-x86_64-static
Downloading from: https://github.com/commercialhaskell/stack/releases/download/v2.3.1/stack-2.3.1-linux-x86_64-static.tar.gz
Download complete, testing executable
Version 2.3.1, Git revision de2a7b694f07de7e6cf17f8c92338c16286b2878 (8103 commits) x86_64 hpack-0.33.0

Warning: Installation path /root/.local/bin
         not found on the PATH environment variable.
New stack executable available at /root/.local/bin/stack
Also copying stack executable to /usr/bin/stack
Stack executable copied successfully!
[schwarzgerat](0) $
```

ok, despite it claiming to have written to `/root/.local/bin/stack` and `/usr/bin/stack`, i also somehow seem to have my own `.local` copy with the correct version(?):

```
[schwarzgerat](130) $ stack --version
Version 2.3.1, Git revision de2a7b694f07de7e6cf17f8c92338c16286b2878 (8103 commits) x86_64 hpack-0.33.0
[schwarzgerat](0) $ which stack
/home/dank/.local/bin/stack
[schwarzgerat](0) $ 
```

and....yes, the updated `stack` appears to pull `github-release` as advertised:

```
Building all executables for `github-release' once. After a successful build of all of them, only specified executables will be rebuilt.
github-release       > configure (lib + exe)
Configuring github-release-1.3.4...
github-release       > build (lib + exe)
Preprocessing library for github-release-1.3.4..
Building library for github-release-1.3.4..
[1 of 2] Compiling Paths_github_release
[2 of 2] Compiling GitHubRelease
Preprocessing executable 'github-release' for github-release-1.3.4..
Building executable 'github-release' for github-release-1.3.4..
[1 of 1] Compiling Main       
Linking .stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.0.0/build/github-release/github-release ...
                              
Warning: Failed to decode module interface:
         /home/dank/src/github-release/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.0.0/build/GitHubRelease.hi
         Decoding failure: Invalid magic: e49ceb0f
                              
Warning: Failed to decode module interface:
         /home/dank/src/github-release/.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.0.0/build/github-release/github-release-tmp/Main.hi
         Decoding failure: Invalid magic: e49ceb0f
github-release       > copy/register
Installing library in /home/dank/src/github-release/.stack-work/install/x86_64-linux-tinfo6/343950e87868e4c366c7751e5804b835291ac910ca39398c4e407d0f6264917e/8.10.1/lib/x86_64-linux-ghc-8.10.1/github-release-1.3.4-EBco0GEEjGh4YlOs4O8TtY
Installing executable github-release in /home/dank/src/github-release/.stack-work/install/x86_64-linux-tinfo6/343950e87868e4c366c7751e5804b835291ac910ca39398c4e407d0f6264917e/8.10.1/bin
Registering library for github-release-1.3.4..
Completed 58 action(s).       
Copying from /home/dank/src/github-release/.stack-work/install/x86_64-linux-tinfo6/343950e87868e4c366c7751e5804b835291ac910ca39398c4e407d0f6264917e/8.10.1/bin/github-release to /home/dank/.local/bin/github-release

Copied executables to /home/dank/.local/bin:
- github-release
```

ok, thanks! I'm closing this issue; if you want to try to support older `stack` and would like me to test, feel free to reopen it.

**Comment by tfausak** [comment]
The problem with the older version of Stack isn't `github-release` exactly, it's the snapshot it's trying to use. You could try using a snapshot that was released around the time of Stack 1.9.3, like LTS 12. For example:

``` sh
stack --resolver lts-12 install github-release
```

But don't feel like you need to. I made this package for people that would prefer Haskell-powered tooling. If another similar project works for you, that's great!

**Comment by dankamongmen** [comment]
I can confirm that your example allows me to run `github-release` on a Debian Unstable workstation, w00t!

And yeah I just grabbed the second tool off the list =]. I wanted to help you chase down any bug that might have existed, though. Always happy to play with Haskell.
